File: problems/#629/v2.cpp
class Solution {
public:
    int kInversePairs(int n, int k) {
        int MOD = 1e9 + 7, bound = 0;
        vector<vector<int>> dp(n + 1, vector<int>(k + 1, 0));
        for (int i = 1; i <= n; ++i) {
            dp[i][0] = 1;
            bound = min(bound + i - 1, k);
            for (int j = 1; j <= bound; ++j) {
                dp[i][j] = (dp[i][j - 1] + dp[i - 1][j]) % MOD;
                if (j - i >= 0) {
                    dp[i][j] = (dp[i][j] - dp[i - 1][j - i] + MOD) % MOD;
                }
            }
        }
        return dp[n][k];
    }
};

File: problems/#2466/v2.go
func countGoodStrings(low int, high int, zero int, one int) int {
    MOD := int(1e9 + 7)
    ret, bound, dp := 0, max(high - zero, high - one) + 1, make([]int, high + max(zero, one))
    dp[0] = 1
    for i := range bound {
        dp[i + zero] = (dp[i + zero] + dp[i]) % MOD
        dp[i + one] = (dp[i + one] + dp[i]) % MOD
    }
    for i := low; i <= high; i++ {
        ret = (ret + dp[i]) % MOD
    }
    return ret
}